kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -458,12 +458,10 @@ protected ApiVersionsResponse overloadDefaultApiVersionsResponse(boolean unsuppo
             for (ApiKeys apiKey : ApiKeys.values()) {
                 if (apiKey.minRequiredInterBrokerMagic <= RecordBatch.CURRENT_MAGIC_VALUE) {
                     switch (apiKey) {
-                        case LIST_OFFSETS:
-                            // V0 is needed for librdkafka
-                            versionList.add(new ApiVersion((short) 2, (short) 0, apiKey.latestVersion()));
-                            break;
-                        default:
-                            versionList.add(new ApiVersion(apiKey));
+                        case LIST_OFFSETS ->
+                                // V0 is needed for librdkafka
+                                versionList.add(new ApiVersion((short) 2, (short) 0, apiKey.latestVersion()));
+                        default -> versionList.add(new ApiVersion(apiKey));
```

Comment:
We'd better still keep the code compatible with Java 8 for existing code so that we can cherry-pick the PR into older branches.

BTW, it looks like the JavaDocs plugin still uses an older source as well.

```
Error:  /home/runner/work/kop/kop/kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java:461: error: switch rules are not supported in -source 8
Error:                          case LIST_OFFSETS ->
Error:                                            ^
Error:    (use -source 14 or higher to enable switch rules)
```